Update the way parallelization is configured

dkuegler opened the pull request:
- Update the `--parallel_subjects*` flags and rename them to `--parallel*` (`brun_fastsurfer.sh`, `long_fastsurfer.sh`)
- Update the "old" `--parallel` flag (making it legacy, removing from documentation everywhere incl. all examples, but also from the "functionality of the scripts")
- Update the way `--threads` works for surface reconstruction. Before: `run_fastsurfer.sh` fixes `--threads` value, if `--parallel` is set; Now: `run_fastsurfer.sh` does nothing to the threads value for surface reconstruction, but recon-surf.sh has 2 values for threads: for joint parts and for parallel hemisphere parts; if threads is >= 2 parallel hemispheres is automatically activated and the number of threads available to each parallel hemisphere is halved.

- Update the documentation throughout.

dkuegler committed:
Update the way parallelization is configured

- Update the --parallel_subjects* flags and rename them to --parallel* (brun_fastsurfer.sh, long_fastsurfer.sh)
- Update the "old" --parallel flag (making it legacy, removing from documentation everywhere incl. all examples, but also from the "functionality of the scripts")
- Update the way --threads works for surface reconstruction. Before: run_fastsurfer.sh fixes --threads value, if --parallel is set; Now: run_fastsurfer.sh does nothing to the threads value for surface reconstruction, but recon-surf.sh has 2 values for threads: for joint parts and for parallel hemisphere parts; if threads is >= 2 parallel hemispheres is automatically activated and the number of threads available to each parallel hemisphere is halved.

- Update the documentation throughout.

dkuegler committed:
Fix link to longitudinal processing

m-reuter commented on Docker/README.md:
```diff
@@ -27,8 +27,7 @@ docker run --gpus all -v /home/user/my_mri_data:/data \
                       --t1 /data/subjectX/t1-weighted.nii.gz \
-                      --sid subjectX --sd /output \
-                      --parallel
+                      --sid subjectX --sd /output
```
Don't we have to put threads to 2 in order to have parallel behaviour?

m-reuter replied:
The same further below.

dkuegler replied:
Definitely, I mean, we can also put 2 threads as the default ....

I just removed the --parallel flags because those are not really wanted/needed any more.

m-reuter commented on doc/scripts/BATCH.md:
````diff
@@ -34,49 +34,48 @@ Parallelization with `brun_fastsurfer.sh`
 
 ```bash
-$FASTSURFER_HOME/brun_fastsurfer.sh --parallel_subjects 4 --threads 2 --parallel
+$FASTSURFER_HOME/brun_fastsurfer.sh --parallel 4 --threads 2
````
for same behaviour as before threads would need to be 4?

dkuegler replied:
Yes, but the text below is updated to reflect the behavior as "requested".

m-reuter commented on run_fastsurfer.sh:
```diff
@@ -535,6 +537,19 @@ check_allow_root
+  echo "WARNING: The --parallel flag is obsolete and will be removed in FastSurfer 3."
+  echo "  Hemispheres are now automatically processed in parallel, if threads for surface "
+  echo "  reconstruction are more than 1 (defined via --threads 2 or --threads_surf 2)!"
+  echo "IMPORTANT NOTE: The threads behavior has also changed, --parallel used to define the"
```
```suggestion
  echo "IMPORTANT NOTE: The threads behavior has also changed, --threads used to define the"
```

m-reuter committed:
Update run_fastsurfer.sh

dkuegler committed:
Create a temporary logfile for log messages that are generated before the logfile exists.
Append those messages to the logfile once the logfile is created. Cleanup of the temporary logfile.
(Don't care if the temporary logfile gets messages, but run_fastsurfer.sh is terminated because of a different error. It is only a temporary file that is very small.)

dkuegler committed:
Update the documentation of the --threads flag in FLAGS.md, EXAMPLES.md and Docker/README.md. Also remove some outdated information in Docker/README.md.
